src/MessageSigner/MessageSigner.php

```diff
@@ -60,11 +60,11 @@ public function calculateMessageHash(NetworkInterface $network, string $message)
 
     /**
      * @param SignedMessage $signedMessage
-     * @param PayToPubKeyHashAddress $address
+     * @param Address $address
      * @param NetworkInterface|null $network
      * @return bool
      */
-    public function verify(SignedMessage $signedMessage, PayToPubKeyHashAddress $address, NetworkInterface $network = null): bool
+    public function verify(SignedMessage $signedMessage, Address $address, NetworkInterface $network = null): bool
```

Comment:
To my knowledge there's no spec for signing P2SH, V1 segwit addresses, V0 segwit script-hash addresses, so this would need a check that the address type is P2PKH or P2WPKH `(V0 segwit && hash->getSize() == 20)`

Reply:
To be honest, I'm a bit new to this. I just noticed that the test wallet I use (Embit Wallet) allows signing of segwit addresses (P2WPKH with a tb1 prefix), tried my changes to this project and saw that it worked. Apparently, both use the same signature algorithm.
I'll look into the verification.

Reply:
I looked into this a bit further. If I figured it out correctly, BIP322 says that messages can only be signed with bech32 segwit addresses of v0 (besides legacy P2PKH). To check this in the verify method, I'm going to verify that the address is either a PayToPubKeyHashAddress object, or, if it is a SegwitAddress, that the witness program version equals 0.
Wrapped segwit addresses are represented as ScriptHashAddress, so they will fall out too.

Reply:
I tried to create a test case for a v1+ address, but it looks like the AddressCreator doesn't handle bech32m, at least no v1 address I could find on-line is recognized. So for now, the MessageSigner will only sign P2PKH or Segwit v0.